include/utils/utils.hpp

```diff
@@ -36,27 +36,80 @@
 #include <stdarg.h>
 #include <string.h>
 #include <string>
+#include <type_traits>
 #include <vector>
 
+namespace NumberFormatUtils
+{
+    // Binary-Coded Decimal (BCD) conversions
+    template <typename T>
+    constexpr T BigEndianBCDtoUInteger(const u8* src, size_t arrayLength)
+    {
+        static_assert(std::is_integral_v<T> && std::is_unsigned_v<T>);
+        T out = 0;
+        T j = 1;
+        for (int i = arrayLength - 1; i >= 0; i--, j *= 100)
+        {
+            out += ((src[i] & 0x0F) * j) + (((src[i] & 0xF0) >> 4) * (j * 10));
+        }
+        return out;
+    }
+    template <typename T, size_t arrayLength = sizeof(T)>
+    constexpr std::array<u8, arrayLength> UIntegerToBigEndianBCD(T src)
```

Comment:
This should have an overload that writes directly out to a u8 pointer (still with an arrayLength template parameter); look at the way that LittleEndian::convertFrom with an array works for an example of what I would prefer